python/demo/demo_static-condensation.py

```diff
@@ -54,15 +55,12 @@
 infile.close()
 
 # Stress (Se) and displacement (Ue) elements
-Se = element("DG", msh.basix_cell(), 1, shape=(2, 2), symmetry=True)
-Ue = element("Lagrange", msh.basix_cell(), 2, shape=(2,))
+Se = element("DG", msh.basix_cell(), 1, shape=(2, 2), symmetry=True, dtype=PETSc.RealType)  # type: ignore
+Ue = element("Lagrange", msh.basix_cell(), 2, shape=(2,), dtype=PETSc.RealType)  # type: ignore
```

Comment:
Do we want to use `dolfinx.default_real_type` here instead of `PETSc.RealType`?

Reply:
This seems like a broader question; if a demo uses PETSc for the final solve, should we be recommending the use of PETSc types throughout?

Reply:
I think that makes sense. For the tests, I think we should ultimately aim to make them PETSc-free unless they are actually testing PETSc integration.

Reply:
This demo uses PETSc solvers and matrices directly, so `dtype=PETSc.RealType` is appropriate in this case.

Ideally, in the end we can remove `dolfinx.default_real_type` and use `PETSc.RealType` where PETSc is explicitly being used.

Reply:
Makes sense.